### ISSUE TITLE
Post-deduplication steps only run on single sample on multi-sample run

### ISSUE THREAD

**Issue by jfy133** [nf-core/eager#140]
**Describe the bug**
When running with multiple samples, after Deduplication, DamageProfiler and Qualimap are only run on for a single sample (note that I skipped preseq)

**To Reproduce**

```
nextflow run nf-core/eager \
-c /projects1/users/fellows/nextflow/nextflow_custom.config \
-profile shh_custom \
--reads '/projects1/microbiome_calculus/RIII/03-preprocessing/pairment_splitting/raw_se/*/*_R1_*.fastq.gz' \
--singleEnd \
--fasta '/projects1/microbiome_calculus/RIII/01-data/reference_genomes/HG19/hg19_complete.fasta' \
--bwa_index '/projects1/microbiome_calculus/RIII/01-data/reference_genomes/HG19/' \
--seq_dict '/projects1/microbiome_calculus/RIII/01-data/reference_genomes/HG19/hg19_complete.dict' \
--outdir '/projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/raw_se/' \
--name 'RIII_pe_raw' \
--email fellows@shh.mpg.de \
--max_cpus 8 \
--max_mem '64.GB' \
--skip_preseq \
--min_adap_overlap 1 \
--clip_readlength 30 \
--clip_min_read_quality 20 \
--bwaalnn 0.01 \
--bwaalnl 32 \
--bam_discard_unmapped \
--bam_unmapped_type fastq \
--dedupper dedup \
--dedup_all_merged \
-resume \
-r 2.0.5
```

**Expected behavior**

Files for all samples generated in DeDup to be run through Qualimap/DamageProfiler

**Screenshots**

You can see here under DamageProfiler only a single sample, despite in DeDuplication all four samples are procssed

```
.
├── damageprofiler
│   └── SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted
│       ├── 3p_freq_misincorporations.txt
│       ├── 3pGtoA_freq.txt
│       ├── 5pCtoT_freq.txt
│       ├── 5p_freq_misincorporations.txt
│       ├── DamagePlot.pdf
│       ├── DamageProfiler.log
│       ├── dmgprof.json
│       ├── DNA_comp_genome.txt
│       ├── DNA_composition_sample.txt
│       ├── identity_histogram.pdf
│       ├── Length_plot.pdf
│       ├── lgdistribution.txt
│       └── misincorporation.txt
├── deduplication
│   └── dedup
│       ├── dedup.log -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/c2/f7c8c24f2d73efe4a1abbd977f448a/dedup.log
│       ├── ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/c2/f7c8c24f2d73efe4a1abbd977f448a/ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist
│       ├── ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/c2/f7c8c24f2d73efe4a1abbd977f448a/ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam
│       ├── ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/c2/f7c8c24f2d73efe4a1abbd977f448a/ELSIDRON1_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai
│       ├── ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/e6/b735964cec5775d49d35d5a267c1a8/ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist
│       ├── ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/e6/b735964cec5775d49d35d5a267c1a8/ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam
│       ├── ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/e6/b735964cec5775d49d35d5a267c1a8/ELSIDRON2_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai
│       ├── SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/a7/cb2f1f9edbf7f4b6f9b49fa4cf543d/SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist
│       ├── SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/a7/cb2f1f9edbf7f4b6f9b49fa4cf543d/SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam
│       ├── SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/a7/cb2f1f9edbf7f4b6f9b49fa4cf543d/SPYNEW_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai
│       ├── SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/1c/07ac33303da3549468f844b3708142/SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.hist
│       ├── SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/1c/07ac33303da3549468f844b3708142/SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam
│       └── SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai -> /projects1/microbiome_calculus/RIII/03-preprocessing/sequencing_qc/preprocessed_se/work/1c/07ac33303da3549468f844b3708142/SPYOLD_S0_L001_R1_001.fastq.merged.combined.sorted.bam.filtered.sorted.bam.bai

```

**Additional context**
Please provide me with the following files:
[nextflow.log](https://github.com/nf-core/eager/files/2824466/nextflow.log)
[eager_report.html.txt](https://github.com/nf-core/eager/files/2824470/eager_report.html.txt)
[eager_timeline.html.txt](https://github.com/nf-core/eager/files/2824471/eager_timeline.html.txt)
[eager_trace.txt](https://github.com/nf-core/eager/files/2824467/eager_trace.txt)

**Comment by apeltzer** [comment]
I can see whats the problem. Same as with the indices in the beginning, need to do similar things as the reference FastA is "consumed" in these cases too ...

**Comment by apeltzer** [comment]
Fixed in #151
